include/secp256k1_extrakeys.h

```diff
@@ -0,0 +1,232 @@
+ *  In:        pubkey: pointer to a public key that is converted (cannot be NULL)
+ */
+SECP256K1_API SECP256K1_WARN_UNUSED_RESULT int secp256k1_xonly_pubkey_from_pubkey(
+    const secp256k1_context* ctx,
+    secp256k1_xonly_pubkey *xonly_pubkey,
+    int *pk_parity,
+    const secp256k1_pubkey *pubkey
+) SECP256K1_ARG_NONNULL(1) SECP256K1_ARG_NONNULL(2) SECP256K1_ARG_NONNULL(4);
+
+/** Tweak an x-only public key by adding tweak times the generator to it.
+ *
+ *  Note that because the resulting point may have an odd Y coordinate, it can
+ *  not be represented by an x-only pubkey. Instead, the output_pubkey is a
+ *  normal secp256k1_pubkey.
+ *
+ *  Returns: 0 if the arguments are invalid or the resulting public key would be
+ *           invalid (only when the tweak is the negation of the corresponding
+ *           secret key). 1 otherwise.
+ *
+ *  Args:           ctx: pointer to a context object initialized for validation
+ *                       (cannot be NULL)
+ *  Out:  output_pubkey: pointer to a public key to store the result (cannot be
+ *                       NULL)
+ *  In: internal_pubkey: pointer to an x-only pubkey to apply the tweak to. Will
+ *                       be set to an invalid value if this function returns 0
+ *                       (cannot be NULL).
+ *              tweak32: pointer to a 32-byte tweak. If the tweak is invalid
+ *                       according to secp256k1_ec_seckey_verify, this function
+ *                       returns 0. For uniformly random 32-byte arrays the
+ *                       chance of being invalid is negligible (around 1 in
+ *                       2^128) (cannot be NULL).
+ */
+SECP256K1_API SECP256K1_WARN_UNUSED_RESULT int secp256k1_xonly_pubkey_tweak_add(
+    const secp256k1_context* ctx,
+    secp256k1_pubkey *output_pubkey,
+    const secp256k1_xonly_pubkey *internal_pubkey,
+    const unsigned char *tweak32
+) SECP256K1_ARG_NONNULL(1) SECP256K1_ARG_NONNULL(2) SECP256K1_ARG_NONNULL(3);
+
+/** Tests that output_pubkey is the result of calling
+ *  secp256k1_xonly_pubkey_tweak_add with internal_pubkey and tweak32.
+ *
+ *  The output_pubkey is represented by its 32-byte x-only serialization and its
+ *  pk_parity, which can both be obtained by converting the result of tweak_add
+ *  to a secp256k1_xonly_pubkey.
+ *
+ *  Note that this alone does _not_ verify that output_pubkey is a commitment.
+ *  If the tweak is not chosen in a specific way, the output_pubkey can easily
+ *  be the result of a different internal_pubkey and tweak.
+ *
+ *  Returns: 0 if the arguments are invalid or the output_pubkey is not the
+ *           result of tweaking the internal_pubkey with tweak32. 1 otherwise.
+ *  Args:           ctx: pointer to a context object initialized for validation
+ *                       (cannot be NULL)
+ *  In: output_pubkey32: pointer to a serialized xonly_pubkey (cannot be NULL)
+ *     output_pk_parity: the pk_parity value that is returned when
```

Comment:
micro-nitting: Then naming scheme for this arg seems very taproot-specific.

Reply:
I don't see how the naming is taproot-specific. You could also call it odd_y or something but that's unnecessarily technical. The thing I want to convey is that this belongs to the `output_pk`.

Reply:
Oh sorry, I meant `output_*` as a whole.

Reply:
Oh right! Perhaps that holds even more for `internal_pubkey`. But given what `tweak_add` does, I think those names are okay even outside of the taproot context with the benefit that they're more understandable than alternatives *in* the taproot context.